File: leetcode/medium/2348. Number of Zero-Filled Subarrays/main.go
package main

func zeroFilledSubarray(nums []int) int64 {
	current_zero_subarray, all_zero_subarrays := int64(0), int64(0)

	for _, num := range nums {
		if num == 0 {
			current_zero_subarray++
			all_zero_subarrays += current_zero_subarray
		} else {
			current_zero_subarray = 0
		}
	}

	return all_zero_subarrays
}

File: leetcode/medium/2348. Number of Zero-Filled Subarrays/main.ts
const zeroFilledSubarray = (nums: number[]): number => {
  let [currentZeroSubarray, allZeroSubarrays] = [0, 0]

  for (const num of nums) {
    if (num == 0) {
      currentZeroSubarray++
      allZeroSubarrays += currentZeroSubarray
    } else {
      currentZeroSubarray = 0
    }
  }

  return allZeroSubarrays
}
